Dockerfile

```diff
@@ -0,0 +1,24 @@
+# Miden-node Dockerfile
+
+#### Setup image builder
+FROM rust:1.75.0-bullseye AS builder
+
+# Install dependencies
+RUN apt-get update && apt-get -y upgrade
+RUN apt-get install -y gcc llvm clang bindgen pkg-config
```

Comment:
please document why these dependencies are necessary. Something along the lines of:

```
# Require to build sqlite: gcc llvm clang bindgen pkg-config
```

Also, why do we need both gcc and clang?

Reply:
I have followed what was written in the README:

https://github.com/0xPolygonMiden/miden-node/blob/3c3ebbecbece73b330fe01cfa2b6a41c825c8db1/README.md?plain=1#L35

Indeed I don't think that we need both, should we make a modification?

Reply:
@polydez do we need both compilers? If not, I would simplify this.

Reply:
Removed gcc and kept clang

Reply:
@hackaugusto @phklive can't answer without testing, but might work.